app/services/EventReportService.scala

```diff
@@ -433,8 +435,8 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
     Future.fromTry(jsonPayloadSchemaValidator.validatePayload(payload, schemaPath, eventName)).map(_ => (): Unit)
   }
 
-  def submitEventDeclarationReport(pstr: String, userAnswersJson: JsValue, version: String)
-                                  (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Unit] = {
+  def submitEventDeclarationReport(pstr: String, psaPspId: String, userAnswersJson: JsValue, version: String)
+                                  (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Result] = {
```

Comment:
Can we change the return type to Future[Unit] for submitEventDeclarationReport - I don't want Result to be return anywhere but the controller, since response code should be handled via controller.

Ideally we would want to return and process the API response too.

Reply:
After discussion with Pavel, these comments will be addressed in [PODS-8697](https://jira.tools.tax.service.gov.uk/browse/PODS-8697) as this is a substantial refactor which will require QA testing again.